packages/common/src/utils/nft.utils.ts

```diff
@@ -173,4 +186,89 @@ export class NftUtils {
       throw new Error(`onNewNftEvent lambda invoke failed for tx: ${txId}`);
     }
   }
+
+  /**
+   * Process a new NFT event by transforming the data, checking if it should invoke the NFT handler,
+   * and invoking the lambda if appropriate.
+   */
+  static async processNftEvent(
+    eventData: FullNodeTransaction,
+    stage: string,
+    network: unknown,
+    logger: Logger
+  ): Promise<boolean> {
+    // Early return if NFT auto review is disabled
+    if (!isNftAutoReviewEnabled()) {
+      logger.debug('NFT auto review is disabled. Skipping NFT handler invocation.');
+      return false;
+    }
+
+    // Early return if not a token creation transaction
+    if (eventData.version !== constants.CREATE_TOKEN_TX_VERSION) {
+      logger.debug(`Transaction version ${eventData.version} is not a token creation transaction (${constants.CREATE_TOKEN_TX_VERSION}). Skipping NFT handler invocation.`);
+      return false;
+    }
+
+    try {
+      // Transform the data to a format compatible with shouldInvokeNftHandlerForTx
+      const transformedTx = NftUtils.transformFullNodeTxForNftDetection(eventData);
+
+      // Check if we should invoke the NFT handler for this transaction
+      if (NftUtils.shouldInvokeNftHandlerForTx(transformedTx, network, logger)) {
+        // Get the transaction hash
+        const txId = eventData.hash;
+
+        // Invoke the lambda function
+        await NftUtils.invokeNftHandlerLambda(txId, stage, logger);
+        return true;
+      }
+
+      return false;
+    } catch (error) {
+      logger.error(`Error processing NFT event: ${error}`);
+      return false;
+    }
+  }
+
+  /**
+   * Transform transaction data from the full node to a format compatible with the NFT detection logic.
+   */
+  static transformFullNodeTxForNftDetection(fullNodeData: FullNodeTransaction): HistoryTransaction {
+    // Create a new object with the required properties
+    const transformedTx: any = {
+      hash: fullNodeData.hash,
+      tx_id: fullNodeData.hash, // Add tx_id for compatibility
+      version: fullNodeData.version,
+      tokens: fullNodeData.tokens,
+      token_name: fullNodeData.token_name,
+      token_symbol: fullNodeData.token_symbol,
+      inputs: fullNodeData.inputs.map((input: FullNodeInput) => {
+        const tokenIndex = (input.spent_output.token_data & constants.TOKEN_INDEX_MASK) - 1;
```

Comment:
suggestion(non-blocking): Low level operations like bit manipulations are always best accompanied by a comment explaining them, as they are harder to read than common javascript code.